scripts/notebook_text_checker.py

```diff
@@ -517,6 +517,18 @@ def file_list(path, errors):
 
 This check can be run locally, via `{command}`."""
 
+        if "GITHUB_ACTIONS" in os.environ:
+
+            def github_file_list(path, errors):
+                whole_list = "%0A".join(error.replace("\n", "%0A") for error in errors)
```

Comment:
Another option here would be to make it more "human-facing" for writing, and then do a single escaping step at the end; something like:

```suggestion
                whole_list = "\n".join(error for error in errors)
                message = f"""\
Notebook failed text check

{whole_list}"""
                escaped = message.replace("\n", "%0A")
```

This also makes it easier to incorporate more text, e.g. the `command`, if you think that's sensible.

This is totally optional and ties into the suggestion below about doing one loop.